lib/pubsueque/launcher.rb

```diff
@@ -45,6 +45,12 @@ def init_subscribers
         subscription.listen(subscription_options) do |message|
           if schedule?(message)
             processor = Processor.new(message, @retry_processor, @stats, @options)
+            # We do this elsewhere as well, extending deadlines.
+            # (processor#retry_job)
+            # Can it be centralized?
+            # Why do we need the `Scheduler`?
+            # We could just give the message back to GCP until it's ready to
+            # be processed.
```

Comment:
You can delay messages by up to ten minutes, after which the message will be redelivered. It's means at least every ten minutes a message may have to be re-delayed.

Reply:
Oh Okay great. This delaying messages, is it by configuring the retry_policy minimum/maximum_backoff?

Reply:
I don't exactly remember how the retry policy works, but one thing you can do is to dequeue, check if the message is ready to be processed, otherwise modify the ack deadline and ignore it until it's redelivered.